pkg/datastore/pubsub/kafka/kafka.go

```diff
@@ -205,6 +207,63 @@ func New(config *Config, logger log.Logger) (*Kafka, error) {
 	return kafkaObj, nil
 }
 
+// NewKafka establishes connection to Kafka and enables AVRO using the provided configs
+func NewKafka(kafkaConfig *Config, avroConfig *avro.Config, logger log.Logger) (*Kafka, error, error) {
+	pubsub.RegisterMetrics()
+
+	if kafkaConfig.SASL.Mechanism != SASLTypeSCRAMSHA512 && kafkaConfig.SASL.Mechanism != PLAIN && kafkaConfig.SASL.User != "" {
+		return nil, errInvalidMechanism, nil
+	}
+
+	populateOffsetTopic(kafkaConfig)
+	convertKafkaConfig(kafkaConfig)
+
+	sarama.Logger = kafkaLogger{logger: logger}
+
+	kafkaObj, err := initializePubSubInstance(kafkaConfig, logger)
+	if err != nil {
+		return &Kafka{config: kafkaConfig, logger: logger}, err, nil
+	}
+
+	logger.Infof("Kafka initialized. Hosts: %v, Topics: %v\n", kafkaConfig.Brokers, kafkaConfig.Topics)
+
+	a, err := avro.NewClient(avroConfig)
+	if err != nil {
+		return kafkaObj, nil, err
+	}
+
+	if a != nil {
+		kafkaObj.avro = a
+
+		logger.Infof("Avro initialized! SchemaRegistry: %v SchemaVersion: %v, Subject: %v",
+			avroConfig.URL, avroConfig.Version, avroConfig.Subject)
+	}
+
+	return kafkaObj, nil, nil
+}
+
+// initializePubSubInstance initialize the sarama producer and consumer instances on the basis of provided configs
+func initializePubSubInstance(cfg *Config, logger log.Logger) (*Kafka, error) {
+	brokers := strings.Split(cfg.Brokers, ",")
+
+	p, err := sarama.NewSyncProducer(brokers, cfg.Config)
+	if err != nil {
+		return &Kafka{config: cfg, logger: logger}, err
```

Comment:
In these error cases from `NewSyncProducer` or `NewKafkaConsumer`, can we return nil instead of that Kafka Object?